cmd/ci-operator-config-mirror/main.go

```diff
@@ -51,6 +56,40 @@ func (o *options) validate() error {
 	return o.WhitelistOptions.Validate()
 }
 
+type configsByRepo map[string][]config.DataWithInfo
+
+func (d configsByRepo) cleanDestinationSubdirs(destinationOrgPath string) error {
+	var subDirs []string
+
+	if err := filepath.Walk(destinationOrgPath, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() && path != destinationOrgPath {
+			subDirs = append(subDirs, path)
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("Failed to walk through the destination dir: %v", err)
+	}
+
+	for _, subDir := range subDirs {
+		if err := os.RemoveAll(subDir); err != nil {
+			return fmt.Errorf("couldn't delete dir: %v", err)
+		}
+	}
```

Comment:
Looks unnecessarily complicated, the `Walk` is not actually needed (it implies there is a hierarchy, but there is really just a single level of subdirs). It should be enough to `ioutil.ReadDir` and iterate over that, skipping everything that is not a dir.